core/services/pipeline/runner.go

```diff
@@ -236,133 +300,94 @@ func (r *runner) executeRun(
 		}
 	)
 
-	var taskDAG TaskDAG
-	err := taskDAG.UnmarshalText([]byte(spec.DotDagSource))
+	pipeline, err := Parse([]byte(spec.DotDagSource))
 	if err != nil {
 		return run, nil, false, err
 	}
 
-	headTaskRuns, err := r.memoryTaskRunDAGFromTaskDAG(taskDAG, pipelineInput, txdb)
-	if err != nil {
-		return run, nil, false, err
+	// initialize certain task params
+	txMu := &sync.Mutex{}
+	for _, task := range pipeline.Tasks {
+		if task.Type() == TaskTypeHTTP {
+			task.(*HTTPTask).config = r.config
+		} else if task.Type() == TaskTypeBridge {
+			task.(*BridgeTask).config = r.config
+			task.(*BridgeTask).safeTx = SafeTx{txdb, txMu}
```

Comment:
I actually think we can dump `SafeTx` now since we always pass in a non-transactional database